README.md

```diff
@@ -1,32 +1,289 @@
+programming languages, object orientation will be implemented and structured in
+different ways.
+
+[functional programming language]: https://en.wikipedia.org/wiki/Functional_programming
+
+We can say, though, that our `dog` object is heading in the direction of object
+orientation: our `dog` object represents a real-world thing, and it has
+properties and methods. However, it is limited because it represents a single,
+specific dog. In object orientation, we can create a `Dog` _class_ (with a
+capital "D") that represents dogs in general, and we can then create an
+_instance_ of that class that represents an individual dog. Each dog instance
+would have its own values of the `breed` and `name` properties, and each dog
+instance would be able to `bark()`. The properties and methods of the Dog class
+would be _inherited_ by each instance of the class.
+
+Don't worry too much about functional vs. object-oriented programming at this
+point. These are difficult concepts. We will delve more deeply into what they
+mean later in the course.
+
+## Arrays (and many other things in JavaScript) are Objects
+
+So how does all this apply to JavaScript Arrays? Well, JavaScript Arrays are
+actually objects! They have properties (e.g., `length`) and methods (`push()`,
+`join()`, etc.). Their indexes are actually keys! JavaSCript includes code
+"under the hood" that allows us to interact with Arrays in an array-like way.
+Because all arrays have numeric keys, beginning with 0, that under-the-hood code
+is able to simplify and streamline how we interact with arrays. And when we
+create an array, it _inherits_ all the properties and methods that belong to the
+`Array` object.
+
+So what else in JavaScript is an Object? Lots of things, including strings and
+even functions! In general, we don't need to concern ourselves too much with
+this fact. However, noting the commonalities — e.g., calling methods using dot
+notation — and understanding that instances of these objects inherit all the
+properties and methods of their [object _prototype_][prototypes], can help us
+understand how things work in JavaScript.
+
+## Array Iterator Methods
+
+So now let's return to our Array iterator methods. These are methods on the
+`Array` Object prototype that are inherited by array instances. Therefore, we
+can call any of these iterator methods on a specific array using dot notation.
+
+Like a `for...of` loop, each of the methods _iterates_ through the elements in
+an array in turn and performs some action on each of them. But that's really
+where the similarities end. Unlike a loop, iterator methods:
+
+1. are _called on_ an instance of an array, and
+2. take a _callback function_ as an argument.
+
+For `forEach`, the basic setup looks like this:
+
+```js
+<array>.forEach(callback);
+```
+
+For example, if we want to log the values of our array, we could do this:
+
+```js
+const myArray = [1, 2, 3];
+myArray.forEach(function(el) {
+    console.log(el);
+});
```

Comment:
Maybe provide the output here so they can also see what happens? Suggestion applies to all the samples moving forward as well